src/client/Client.cc

```diff
@@ -8988,6 +8993,24 @@ int Client::_setxattr(Inode *in, const char *name, const void *value,
 int Client::ll_setxattr(Inode *in, const char *name, const void *value,
 			size_t size, int flags, int uid, int gid)
 {
+  // For setting pool of layout, MetaRequest need osdmap epoch.
+  // There is a race which create a new data pool but client and mds both don't have.
+  // Make client got the latest osdmap which make mds quickly judge whether get newer osdmap.
+  if (strcmp(name, "ceph.file.layout.pool") == 0 ||  strcmp(name, "ceph.dir.layout.pool") == 0 ) {
```

Comment:
There is a third case, where the user sets ceph.[file|dir].layout (i.e. sets all parts of layout at once).